apps/frontend/src/hooks/useLoadContract.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { ContractInterface, ethers } from 'ethers';
+
+import { ContractGroup, getContract } from '@sovryn/contracts';
+import { ChainId } from '@sovryn/ethers-provider';
+
+import { defaultChainId } from '../config/chains';
+
+import { useAccount } from './useAccount';
+import { useIsMounted } from './useIsMounted';
+
+export const useLoadContract = (
+  contractName: string,
+  group: ContractGroup,
+  chain: ChainId = defaultChainId,
+  customSigner?: ethers.providers.JsonRpcSigner,
+) => {
+  const isMounted = useIsMounted();
+  const { signer: userSigner } = useAccount();
+
+  const [address, setAddress] = useState<string | undefined>(undefined);
+  const [abi, setAbi] = useState<ContractInterface | undefined>(undefined);
+
+  useEffect(() => {
+    if (isMounted()) {
+      getContract(contractName, group, chain).then(result => {
+        if (isMounted()) {
```

Comment:
First condition prevents getContract from having it being called if component mounts and unmount instantly, second prevents setting to non existing state if getContract was called but was not yet resolved till it was unmounted.

But will remove first condition because getContract caches response and it's not http request so it's actually pretty good it was called